v2/internal/controllers/networking_firewall_20240301_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20240301"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+// Refer to https://learn.microsoft.com/en-us/azure/firewall-manager/quick-firewall-policy-bicep for example firewall configuration
+func Test_Networking_Firewall_20240301_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVNet20240301(tc, testcommon.AsOwner(rg), []string{"10.0.0.0/16"})
+	publicIP := newPublicIP20240301(tc, testcommon.AsOwner(rg))
+	subnet := newSubnet20240301(tc, vnet, "10.0.0.0/24")
+	subnet.Name = "azurefirewallsubnet" // Name must be AzureFirewallSubnet for Azure Firewall
```

Comment:
Slight (casing) inconsistency between the name and the comment - is this maybe case sensitive?